### ISSUE TITLE
Warning/Error while loading from checkpoints after training

### ISSUE THREAD

**Issue by mmaaz60** [zzh8829/yolov3-tf2#108]
Hi,

I trained Tiny Yolo on VOC2012 dataset with model.fit() method using this repo. After training while loading model from checkpoints using detect.py, I am getting warnings that all the checkpoints are not being loaded into the model. Why is it so? The screen shot is attached. 

Note that I did not get any warning when I convert darknet weights using convert.py and load them using detect.py.

Thanks
![WarningWhileLoadingModelFromCheckpointsAfterTraining](https://user-images.githubusercontent.com/55104795/69483380-68636500-0e48-11ea-9932-5d4701e77c20.png)

**Comment by muhammad-maaz-confiz** [comment]
Hi @zzh8829,

Any comments on this thread?

**Comment by PieroCV** [comment]
> Hi,
> 
> I trained Tiny Yolo on VOC2012 dataset with model.fit() method using this repo. After training while loading model from checkpoints using detect.py, I am getting warnings that all the checkpoints are not being loaded into the model. Why is it so? The screen shot is attached.
> 
> Note that I did not get any warning when I convert darknet weights using convert.py and load them using detect.py.
> 
> Thanks
> ![WarningWhileLoadingModelFromCheckpointsAfterTraining](https://user-images.githubusercontent.com/55104795/69483380-68636500-0e48-11ea-9932-5d4701e77c20.png)

I was playing with the code and realised that this warning appears when you train a model with a number of classes and when you load the weights to test the model, the number of classes is different...

**Comment by mmaaz60** [comment]
Hi @PieroCV ,

I am passing --class and --class_num properly in detect.py and still getting this warning. I also noticed that the model building method returns relatively small model while training. Is it something to do with this warning?
![iftraining](https://user-images.githubusercontent.com/55104795/69814375-e0010d80-1215-11ea-87ae-9a31479c7f73.png)

**Comment by mmaaz60** [comment]
Hi All,

Please note that I got this warning when training using model.fit() method. In eager mode training, no such warning. What could have gone wrong?

**Comment by zzh8829** [comment]
This warning is harmless, I made a commit recently to suppress it.
Basically the reason it happens is because model.save_weights includes additional variables from gradient decent optimizer which are not needed during inference. Tensorflow gives warning because there are extra unused variables.

You can disable the warning like i did in `detect.py` by using 
```python
yolo.load_weights(FLAGS.weights).expect_partial()
```
